rtc-service/requests/join-room-request.go

```diff
@@ -18,7 +23,35 @@ func (r *JoinRoomRequest) validate() error {
 }
 
 // Handles the request and returns a response.
-func (r *JoinRoomRequest) Handle(m *Message) (string, error) {
-	// This method will be completed in the future PR.
-	return "Join Room Request", nil
+func (r *JoinRoomRequest) Handle(m *Message, c *websocket.Conn) (string, error) {
+
+  /*
+  Example request
+  {
+    "request-type": "join-room",
+    "data": "{ \"userID\": \"1234\", \"roomID\": \"4321\" }"
+  }
+  */
+
+  // Process request
+  err := json.Unmarshal([]byte(m.Data), &r)
+
+  if err != nil {
+    return "Err", err
+  }
+
+  room := rooms[r.RoomID] 
+
+  // Join room 
+  if &room != nil {
+    room.AddUser(r.UserID, c) 
+    return "Added " + r.UserID + " to " + r.RoomID, nil;
+  } else {
+    // Room doesn't exist -> create room  
+    room = Room{ RoomID: r.RoomID }
```

Comment:
I believe `room` object needs to be persisted back into `rooms` map

Reply:
Fixed